src/LiveSplit.Core/Model/RunFactories/StandardFormatsRunFactory.cs

```diff
@@ -134,6 +134,15 @@ public IRun Create(IComparisonGeneratorsFactory factory)
             }
         }
 
+        using (LiveSplitCore.RunMetadataCustomVariablesIter iter = metadata.CustomVariables())
+        {
+            LiveSplitCore.RunMetadataCustomVariableRef variable;
+            while ((variable = iter.Next()) != null)
+            {
+                run.Metadata.GetOrAddCustomVariable(variable.Name()).AsPermanent().Value = variable.Value();
+            }
+        }
```

Comment:
Use simple `using` statements.

```suggestion
        using var variables = metadata.CustomVariables();
        
        while (variables.Next() is { } variable)
        {
            run.Metadata.GetOrAddCustomVariable(variable.Name()).AsPermanent().Value = variable.Value();
        }
```

I don't know where this stuff comes from, but why are `variable.Name()` and `variable.Value()` methods?

Reply:
I'm using the same style for this `CustomVariables` iteration as was already used for the `SpeedrunComVariables` iteration just above it. Since there are two such iterations, I fear that flattening it out like you propose will make it harder to read, not easier.

The `.Name()` and `.Value()` methods come from the `RunMetadataCustomVariableRef` class in the generated `LiveSplitCore.g.cs` file, a step on the bridge between Rust code and C Sharp code.